Composer/packages/client/src/components/ManageLuis/ManageLuis.tsx

```diff
@@ -0,0 +1,680 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import { useState, useEffect, Fragment } from 'react';
+import { Dialog, DialogType, DialogFooter } from 'office-ui-fabric-react/lib/Dialog';
+import { PrimaryButton, DefaultButton } from 'office-ui-fabric-react/lib/Button';
+import formatMessage from 'format-message';
+import { Icon } from 'office-ui-fabric-react/lib/Icon';
+import { TextField } from 'office-ui-fabric-react/lib/TextField';
+import { Spinner } from 'office-ui-fabric-react/lib/Spinner';
+import { useRecoilValue } from 'recoil';
+import { Dropdown, IDropdownOption } from 'office-ui-fabric-react/lib/Dropdown';
+import { SubscriptionClient } from '@azure/arm-subscriptions';
+import { Subscription } from '@azure/arm-subscriptions/esm/models';
+import { TokenCredentials } from '@azure/ms-rest-js';
+import { CognitiveServicesManagementClient } from '@azure/arm-cognitiveservices';
+import { ResourceManagementClient } from '@azure/arm-resources';
+import { ChoiceGroup, IChoiceGroupOption } from 'office-ui-fabric-react/lib/ChoiceGroup';
+import { ProvisionHandoff } from '@bfc/ui-shared';
+
+import { AuthClient } from '../../utils/authClient';
+import { AuthDialog } from '../../components/Auth/AuthDialog';
+import { armScopes } from '../../constants';
+import { getTokenFromCache, isShowAuthDialog, isGetTokenFromUser } from '../../utils/auth';
+import { LUIS_REGIONS } from '../../constants';
+import { dispatcherState } from '../../recoilModel/atoms';
+
+type ManageLuisProps = {
+  hidden: boolean;
+  onDismiss: () => void;
+  onGetKey: (settings: { authoringKey: string; endpointKey: string; authoringRegion: string }) => void;
+  onNext?: () => void;
+  setDisplayManageLuis: (value: any) => void;
+};
+
+type KeyRec = {
+  name: string;
+  region: string;
+  resourceGroup: string;
+  key: string;
+};
+
+const dropdownStyles = { dropdown: { width: '100%' } };
+const summaryLabelStyles = { display: 'block', color: '#605E5C', fontSize: '14' };
+const summaryStyles = { background: '#F3F2F1', padding: '1px 1rem' };
+const mainElementStyle = { marginBottom: 20 };
+const CREATE_NEW_KEY = 'CREATE_NEW';
+
+const handoffInstructions = formatMessage(
```

Comment:
this needs to be executed in the context of a component, maybe make it a func like getBlahMessage()

Reply:
just moved it inline with the other ui text for this modal.